### PR TITLE
Remove the `box_pointers` lint.

### PR DESCRIPTION
**Pull request by nnethercote**
As the comment says, this lint "is mostly historical, and not particularly useful". It's not worth keeping it around.

r? @estebank

### CODE REVIEW

**Review by lcnr — COMMENTED** [review]
there's a list of removed/renamed lints somewhere to adjust the error message when using a renamed lint. Please add it there

r=me on the impl changes, please start a compiler FCP for this

**Comment by lcnr** [comment]
r? lcnr

**Comment by nnethercote** [comment]
I have added a call to `register_removed`.

**Comment by nnethercote** [comment]
Removing this old, useless lint that tells you if you are using `Box`. (Yes, really. Perhaps it made more sense in pre-1.0 days.)

@rfcbot fcp

**Comment by nnethercote** [comment]
Removing this old, useless lint that tells you if you are using Box. (Yes, really. Perhaps it made more sense in pre-1.0 days.)

@rfcbot fcp merge

**Comment by nnethercote** [comment]
How do I start an FCP? Or does it have to be started by a compiler team member? I'm only a contributor.

**Comment by lcnr** [comment]
@rfcbot fcp merge

File: compiler/rustc_lint/src/lib.rs
Comment (lcnr) [inline compiler/rustc_lint/src/lib.rs:551]:
```suggestion
    store.register_removed(
        "box_pointers", "it was mostly historical and does not detect other kinds of allocations"
    );
```
:blush:

**Review by estebank — APPROVED** [review]

**Comment by lcnr** [comment]
@rfcbot ping

**Comment by lcnr** [comment]
@rfcbot fcp merge

File: compiler/rustc_lint/src/lib.rs
Comment (nnethercote) [reply compiler/rustc_lint/src/lib.rs:551]: I followed the spirit of your suggestion, but ended up using a slightly different wording.

**Comment by nnethercote** [comment]
Does anyone know how to trigger an FCP?

**Comment by kpreid** [comment]
[rfcbot was having some trouble](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/rfcbot.20asleep), since fixed, and it sounds like you should be able to just retry.

**Comment by nnethercote** [comment]
@rfcbot fcp

**Comment by nnethercote** [comment]
@rfcbot fcp merge

**Comment by lcnr** [comment]
@rfcbot fcp merge

**Comment by nnethercote** [comment]
> and at most 2 approvals are outstanding

AFAIK @Aaron1011 and @eddyb have not done any Rust compiler work for some time. Assuming that's right (please correct me if I'm wrong) that means an FCP actually needs approval from all active compiler team members.

**Comment by lcnr** [comment]
they still tend to approve FCPs, even if they currently do not actively contribute as frequently: see https://rfcbot.rs/ for the in-process FCPs

**Review by est31 — APPROVED** [review]

**Comment by nnethercote** [comment]
I rebased.

**Comment by nnethercote** [comment]
> So 👍 to removing it, assuming the correct incantation happened so peoples builds don't break if they were (inexplicably) mentioning it (which I think has already been done in the PR).

Yep, the `register_removed` call takes care of that -- people will get a warning about the lint having been removed.

**Comment by est31** [comment]
Even without `register_removed`, nonexistent lints only trigger `unknown_lints`, which is a warning lint of its own. so it's affected by cargo's lint capping etc.

**Comment by nnethercote** [comment]
Only one approval short of the n-2 threshold!

**Pull request merged**
